pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -687,5 +687,50 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
             arr[k] *= inv_norm;
         }
     }
+
+    /**
+     * @brief Prepares a single computational basis state.
+     *
+     * @param index Index of the target element.
+     */
+    void setBasisState(const std::size_t index) {
+        auto length = this->getLength();
+        PL_ABORT_IF(index > length - 1, "Invalid index");
+
+        auto *arr = this->getData();
+        std::fill(arr, arr + length, 0.0);
+        arr[index] = {1.0, 0.0};
+    }
+
+    /**
+     * @brief Set values for a batch of elements of the state-vector.
+     *
+     * @param values Values to be set for the target elements.
+     * @param indices Indices of the target elements.
+     */
+    void setStateVector(const std::vector<std::size_t> &indices,
+                        const std::vector<ComplexT> &values) {
+        auto num_indices = indices.size();
+        PL_ABORT_IF(num_indices != values.size(),
+                    "Indices and values length must match");
+
+        auto *arr = this->getData();
+        auto length = this->getLength();
+        std::fill(arr, arr + length, 0.0);
+        for (std::size_t i = 0; i < num_indices; i++) {
+            PL_ABORT_IF(i >= length, "Invalid index");
+            arr[indices[i]] = values[i];
+        }
```

Comment:
Do we need to normalize values after updating the indices or validate the new state-vector?

Reply:
We could, but `qml.StatePrep` already validates the norm so I'd say we can skip it.

Reply:
> Do we need to normalize values after updating the indices or validate the new state-vector?

I am not familiar with the code base, so I don't know the answers to this question. I just moved the implementations and addressed the comments. If you find that the answer to this question is a yes, just let me know and we can normalize. I'll keep this conversation open and let you resolve it. :+1:

Reply:
Sounds okay only if we ensure that this function is used from `qml.StatePrep`. I'll approve it now to unblock Erick and we can further discuss it offline :+1: